### PR TITLE
Fix inference/production api to support invocation by task without specifying a card

### PR DESCRIPTION
**Pull request by elronbandel**
From now on you will be able to use `produce` and `infer` to process data by specifying only the task your data points complying to. 

This means you will be able to use any task in the Unitxt catalog as an inference function. 

Key contributions:

1. No need to assign values to output fields in infer/produce functions (only to input fields as the task specify)
2. No need to specify a card unless few shot examples are needed. You can specify the task and it will be enough. 

See the full production.rst for details

### CODE REVIEW

File: docs/docs/production.rst [hunk 1]
```diff
@@ -63,6 +63,18 @@ Then, produce the model-ready input data with the `produce` function:
 
   result = produce(instance, recipe)
 
+If we want to add to the data few shot examples, we should specify recipe with a data card:
+
+.. code-block:: python
+
+  recipe = "card=cards.wnli,template=templates.classification.multi_class.relation.default,demos_pool_size=5,num_demos=2"
```
Comment (OfirArviv) [inline docs/docs/production.rst:70]: I don't understand this example. 
Why are you giving the card of wnli? Will it use the example from there?

File: prepare/processors/to_string.py [hunk 2]
```diff
@@ -1,25 +1,18 @@
 from unitxt import add_to_catalog
 from unitxt.blocks import ToString, ToStringStripped
-from unitxt.operator import SequentialOperator
+from unitxt.processors import PostProcess
+from unitxt.settings_utils import get_constants
+
+constants = get_constants()
```
Comment (OfirArviv) [inline prepare/processors/to_string.py:6]: where is this used? It's in all the files but never get used

File: src/unitxt/api.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ def evaluate(predictions, data) -> List[Dict[str, Any]]:
 
 
 def post_process(predictions, data) -> List[Dict[str, Any]]:
```
Comment (OfirArviv) [inline src/unitxt/api.py:94]: what is this function aim? It's not clear what it does and how the change reflect it

Comment (OfirArviv) [reply src/unitxt/api.py:94]: is it only for inference? What does it means?

File: docs/docs/production.rst (hunk 1 above)
Comment (elronbandel) [reply docs/docs/production.rst:70]: yes the few shot examples come from that card.

File: prepare/processors/to_string.py (hunk 2 above)
Comment (elronbandel) [reply prepare/processors/to_string.py:6]: what is what used?

File: src/unitxt/api.py (hunk 3 above)
Comment (elronbandel) [reply src/unitxt/api.py:94]: This function is for the api to use only post processing on a data without the rest of the pipeline.

File: src/unitxt/processors.py [hunk 4]
```diff
@@ -114,7 +141,7 @@ def process_value(self, text: Any) -> Any:
         return non_empty_line
 
 
-class LowerCase(FieldOperator):
+class Lower(FieldOperator):
```
Comment (eladven) [inline src/unitxt/processors.py:144]: This is a breaking change. We should use gentle deprecation.

File: src/unitxt/schema.py [hunk 5]
```diff
@@ -22,6 +25,38 @@
...
+    {
+        "source": Value("string"),
+        "metrics": Sequence(Value("string")),
+        "groups": Sequence(Value("string")),
+        "subset": Sequence(Value("string")),
+        "postprocessors": Sequence(Value("string")),
+        "task_data": Value(dtype="string"),
+        "data_classification_policy": Sequence(Value("string")),
+    }
+)
```
Comment (eladven) [inline src/unitxt/schema.py:51]:
```suggestion
```

File: src/unitxt/settings_utils.py [hunk 6]
```diff
@@ -87,6 +88,17 @@ def get_all_environment_variables(self):
...
+    def context(self, **kwargs):
+        old_values = {key: self._settings.get(key, None) for key in kwargs}
+        try:
+            for key, value in kwargs.items():
+                self.__setattr__(key, value)
+            yield
+        finally:
+            for key, value in old_values.items():
+                self.__setattr__(key, value)
+
```
Comment (eladven) [inline src/unitxt/settings_utils.py:101]:
```suggestion
```

It should be in a separate PR

**Review by eladven — APPROVED** [review]
